### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by moise-mulungu**
On this milestone, We:
- Created a `spec` folder 
- Used `RSpec` for unit tests
- Added Unit test for: 
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- All classes that you created while implementing the Decorator design pattern
- Updated the README file

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    it 'should have class book' do
+      expect(book.class) == Book
+    end
+
+    it 'should have the correct title and author' do
+      expect(book.title).to eq 'Rising Sun'
+      expect(book.author).to eq 'Dire Akanbi'
+    end
+  end
```
Comment (medaminedev66) [inline spec/book_spec.rb:23]: - [ ] Following the project requirement, you should add unit tests for all the methods in all the classes (as ex: add_rental, add_student, can_use_services?, of_age? ...).

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ciraganenicole  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors
- The tests run successfully

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require './person'
+
+describe 'should create instance of person class' do
+  context 'should initialise person class' do
+    person = Person.new(age: 18, name: 'Daniel', parent_permission: true)
+
+    it 'should be recognized as a person class' do
+      expect(person.class) == Person
+    end
+
```
Comment (KDlamini) [inline spec/person_spec.rb:10]: Please test all methods in each class as stated by the previous reviewer. For instance, even though you have tested the rental class contained in the `rental.rb` file, you must still test the `add_rental` method that in the `person.rb` file in your `person_spec.rb`.

For example:

- In the `person.rb` class you have `add_rental` method, therefore `person.spec.rb` should have a test for `add_rental` similar to this:

```
  context "When adding a new rental" do

    it "The add_rental method returns a rental and adds its to the book" do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Bob")

      rental = person.add_rental("2021-05-04", book)

      expect(person.rentals).to eql [rental]
    end
  end
```

- In the `person.rb` class you also have `can_use_services?` method:
```
  context "When checking permissions" do
    it "The can_use_services? method returns false if under age and no parent permission" do
      person = Person.new(14, "Rose", false)
      expect(person.can_use_services?).to be false
    end
```
    *Also check for positive case scenarios when permission is true

- Now, kindly continue writing tests for the `validate_name` and `check_age` methods too...

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @moise-mulungu 🙋‍♀️,  

You have done an incredible job implementing some of the changes requested by the previous reviewer. 👏🏾👏🏾👏🏾  
However, there are still some changes that aren't addressed yet. You are almost there! 💪🏾 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @ciraganenicole @moise-mulungu 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

ApprovedAgreeGIF


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**